pallets/gear/src/benchmarking/code.rs

```diff
@@ -371,12 +371,24 @@ where
     }
 
     pub fn unary_instr(instr: Instruction, repeat: u32) -> Self {
-        use body::DynInstr::{RandomI64Repeated, Regular};
+        Self::unary_instr_for_arch(instr, true, repeat)
+    }
+
+    pub fn unary_instr_32(instr: Instruction, repeat: u32) -> Self {
+        Self::unary_instr_for_arch(instr, false, repeat)
```

Comment:
I would not be obsessive with primitives: introduce new `enum Arch { Amd64, X86 }` and implement `From` trait for `DynInstr` (or custom function if the latter is impossible).

Reply:
The latter would be nice, but in reality it will require matchiing all the `Instruction` variants which I am not keen on. Will go with a enum as a param instead of bool

Reply:
But even with the enum? Does it worth introducing it in this private code?

Reply:
From my point of view it is worth since improves code readability however I don't insist ).
What concerns conversion function I thought there is no need to match all `Instruction` : it should convert from `Arch` enum to `DynInstr`, something like this:
```rust
fn to_dyn_instr(arch: &Arch, count: i32) -> body::DynInstr {
    match arch {
        Arch::Amd64 => body::DynInstr::RandomI64Repeated(count),
        Arch::X86 => body::DynInstr::RandomI32Repeated(count),
    }
}
```

Reply:
Ahh... I see what you mean. Nice one. Done. Thanks